### PR TITLE
refactor leafletfigures

### PR DESCRIPTION
**Pull request by dhvenema**

### CODE REVIEW

File: src/geonovum/leafletfigures.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import easyButton from "geonovum/deps/easy-button";
+
+export function run(conf, doc, cb) {
+  doc.respecIsReady.then(function() {
+    Array.from(doc.querySelectorAll("figure.scalable img")).forEach(function(
+      image
+    ) {
+      var container = image.parentElement;
+      var div = doc.createElement("div");
+      div.id = "leaflet-figure";
```
Comment (marcoscaceres) [inline src/geonovum/leafletfigures.js]: Bug here... you are generating duplicate IDs in the document.

File: src/geonovum/leafletfigures.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      }).addTo(map);
+      L.easyButton("fa-globe", function() {
+        map.fitBounds(imageBounds);
+      }).addTo(map);
+      L.imageOverlay(image.src, imageBounds).addTo(map);
+      map.fitBounds(imageBounds);
+    });
+  });
+  cb();
+}
```
Comment (marcoscaceres) [inline src/geonovum/leafletfigures.js]: As slightly more idiomatic ES6 implementation of this: 

Using `async/await`, and `const` where appropriate. Makes code a bit cleaner. 

```JS
import "deps/regenerator";
import L from "geonovum/deps/leaflet";
import easyButton from "geonovum/deps/easy-button";

export async function run() {
  await document.respecIsReady;
  document.querySelectorAll("figure.scalable img").forEach(image => {
    const div = document.createElement("div");
    const map = L.map(div, {
      maxZoom: 4,
      minZoom: -4,
      center: [0, 0],
      crs: L.CRS.Simple,
    });
    const { width, height, src } = image;
    const imageBounds = [[0, 0], [height, width]];
    div.classList.add("leaflet-figure");
    image.parentElement.replaceChild(div, image);
    map.setView([height / 2, width / 2], 1);
    L.easyButton("fa-arrows-alt", () => {
      window.open(src, "_blank");
    }).addTo(map);
    L.easyButton("fa-globe", () => {
      map.fitBounds(imageBounds);
    }).addTo(map);
    L.imageOverlay(src, imageBounds).addTo(map);
    map.fitBounds(imageBounds);
  });
}
```

Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: (I've not tested the above)

File: src/geonovum/leafletfigures.js (hunk 1 above)
Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: Here you want: `div.classList.add("leaflet-figure")` ... I'm not sure what setting this ID or class does, however? Do you really need it?

File: src/geonovum/leafletfigures.js (hunk 2 above)
Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: Removed "conf" argument, as it's never used.

Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: Destructured image's props above also - to make things a little simpler.

Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: Slightly more terse implementation: 

```JS
import "deps/regenerator";
import L from "geonovum/deps/leaflet";
import easyButton from "geonovum/deps/easy-button";

export async function run() {
  await document.respecIsReady;
  document.querySelectorAll("figure.scalable img").forEach(image => {
    const { width, height, src } = image;
    const div = document.createElement("div");
    const map = L.map(div, {
      maxZoom: 4,
      minZoom: -4,
      center: [0, 0],
      crs: L.CRS.Simple,
    });
    const imageBounds = [[0, 0], [height, width]];
    map.setView([height / 2, width / 2], 1);
    map.fitBounds(imageBounds);
    [
      L.easyButton("fa-arrows-alt", () => window.open(src, "_blank")),
      L.easyButton("fa-globe", () => map.fitBounds(imageBounds)),
      L.imageOverlay(src, imageBounds),
    ].forEach(item => item.addTo(map));
    div.classList.add("leaflet-figure");
    image.parentElement.replaceChild(div, image);
  });
}
```

File: src/geonovum/leafletfigures.js (hunk 1 above)
Comment (dhvenema) [reply src/geonovum/leafletfigures.js]: The id was for the css, but I should probably use a class attribute for that.

Comment (marcoscaceres) [reply src/geonovum/leafletfigures.js]: IDs should be unique per document, or else you might end up with with strange behaviors (and HTML validation will fail).

File: src/geonovum/leafletfigures.js (hunk 2 above)
Comment (dhvenema) [reply src/geonovum/leafletfigures.js]: If I change the import it works:
```javascript
import * as L from "geonovum/deps/leaflet";
import * as easyButton from "geonovum/deps/easy-button";
```

The page, locally generated, is slow. The generator plugin is not loading:
```
Plugin  took too long. Object { run: run(), 1 meer… }  base-runner.js:51:11
	toRunnable/</</timerId< http://localhost:8080/js/core/base-runner.js:51:11
Error: Plugin  took too long.
Stacktrace:
toRunnable/</</timerId<@http://localhost:8080/js/core/base-runner.js:52:18
  base-runner.js:105:15
	_callee$ http://localhost:8080/js/core/base-runner.js:105:15
	tryCatch http://localhost:8080/js/deps/regenerator.js:64:37
	invoke http://localhost:8080/js/deps/regenerator.js:299:22
	defineIteratorMethods/</prototype[method] http://localhost:8080/js/deps/regenerator.js:116:16
	step http://localhost:8080/js/core/base-runner.js:15:24
	step/< http://localhost:8080/js/core/base-runner.js:28:15
```

Comment (dhvenema) [reply src/geonovum/leafletfigures.js]: I should also change the css to:
```css
.leaflet-container {
    background: #FFF;
    border: 1px solid #8C8C8C;
    width: 100%;
    height: 600px;
    margin-bottom: 1em;
}
```

Comment (dhvenema) [reply src/geonovum/leafletfigures.js]: Also, the first time I load the page I get an error statting `L is not defined`. After a refresh leaflet loads properly.

**Review by marcoscaceres — APPROVED** [review]

**Pull request merged**